### PR TITLE
[shaderwriter] Update to version 2.5.0.

### LINKED ISSUE #27924: [shaderwriter] update to 2.5.0
Library name: ShadeWriter

New version number: 2.5.0

Other information that may be useful (release notes, etc...)

### PR DESCRIPTION
**Pull request by DragonJoker**
**Describe the pull request**

- #### What does your PR fix?
  Fixes #27924

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  No change related to that.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Hopefully yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/shaderwriter/portfile.cmake [hunk 1]
```diff
@@ -1,15 +1,15 @@
 vcpkg_from_github(OUT_SOURCE_PATH SOURCE_PATH
     REPO DragonJoker/ShaderWriter
-    REF ec31f19f88fe15af476b48da7499aac9d4089a8f
+    REF v2.5.0
```
Comment (JonLiu1993) [inline ports/shaderwriter/portfile.cmake:3]:
```suggestion
    REF v${VERSION}
```

File: ports/shaderwriter/portfile.cmake [hunk 2]
```diff
@@ -1,15 +1,15 @@
 vcpkg_from_github(OUT_SOURCE_PATH SOURCE_PATH
```
Comment (JonLiu1993) [inline ports/shaderwriter/portfile.cmake:1]:
```suggestion
vcpkg_minimum_required(VERSION 2022-10-12) # for ${VERSION}

vcpkg_from_github(OUT_SOURCE_PATH SOURCE_PATH
```

File: ports/shaderwriter/portfile.cmake (hunk 1 above)
Comment (JonLiu1993) [reply ports/shaderwriter/portfile.cmake:3]: @DragonJoker, Thanks for your pr, let us use this variable to automatically get the version in vcpkg.json.

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

File: ports/shaderwriter/portfile.cmake (hunk 1 above)
Comment (DragonJoker) [reply ports/shaderwriter/portfile.cmake]: Done.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**